### ISSUE TITLE
Azurite Blob Store not supported due to regex?

### ISSUE THREAD

**Issue by maxhillaert** [GoogleContainerTools/kaniko#1309]
Hi

I assumed kaniko would work with an on-premise blob store using Azurite Blob Storage, but it looks like it's tied to the Cloud Azure Blob Storage domain name

Would it be difficult to support this?  It should be the same API.

**Comment by tejal29** [comment]
yes, @maxhillaert we do support Azure blob with these naming conventions [here](https://github.com/GoogleContainerTools/kaniko/blob/1033ad789b209e20c98f6202e49b292109d92a63/pkg/constants/constants.go#L82) 
It would be easy to update this regular expression and add tests for the newly add expression.

**Comment by dgzlopes** [comment]
I peeked a little on this issue. 

I'm not sure how easy is to update the list of regular expressions, [Azurite Blob Storage](https://github.com/Azure/Azurite) is a self-hosted/on-premise project. 

Any host-suffix can be valid, right? Maybe this check can be disabled with some kind of flag?
